ChainRouter.php

```diff
@@ -249,6 +250,46 @@ public function generate($name, $parameters = array(), $absolute = UrlGeneratorI
         throw new RouteNotFoundException(sprintf('None of the chained routers were able to generate route: %s', $info));
     }
 
+    /**
+     * Rebuild the request object from a URL with the help of the RequestContext.
+     *
+     * If the request context is not set, this simply returns the request object built from $uri.
+     *
+     * @param string $uri
+     *
+     * @return Request
+     */
+    private function rebuildRequest($uri)
+    {
+        if (!$this->context) {
+            return Request::create($uri);
+        }
+
+        $server = array();
+        if ($this->context->getHost()) {
+            $server['SERVER_NAME'] = $this->context->getHost();
+            $server['HTTP_HOST'] = $this->context->getHost();
+        }
+        if ('https' === $this->context->getScheme()) {
+            $server['HTTPS'] = 'on';
+            $server['SERVER_PORT'] = $this->context->getHttpsPort();
+            if (443 !== $this->context->getHttpsPort()) {
+                // this is parsed from the host by symfony request
+                // https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Request.php#L971
+                $server['HTTP_HOST'] .= ':'.$this->context->getHttpsPort();
```

Comment:
@tobion this surprised me - there are really many interactions in the Request object with priorities between different headers and options to set things explicitly...

do you see things i am getting wrong here? like maybe problems with the base path? is there somewhere something already existing to recreate a Request from RequestContext? i only found the other way round, RequestContext::fromRequest.